lib/jekyll/site.rb

```diff
@@ -73,10 +73,11 @@ def process
       render
       cleanup
       write
-      print_stats
+      print_stats if @config["profile"]
     end
 
     def print_stats
+      # TODO: remove duplicate guard statement in next major version change
       if @config["profile"]
```

Comment:
Yes, I didn't want the method to change behavior in a minor release-cycle. Hence the `TODO` comment.